core/src/main/java/io/aiven/klaw/helpers/db/rdbms/InsertDataJdbc.java

```diff
@@ -361,10 +361,10 @@ public boolean hasSequence(String entityName, int tenantId) {
   }
 
   public Integer getNextSeqIdAndUpdate(String entityName, int tenantId) {
-    List<KwEntitySequence> kwEntitySequenceList =
-        kwEntitySequenceRepo.findAllByEntityNameAndTenantId(entityName, tenantId);
-    if (!kwEntitySequenceList.isEmpty()) {
-      Integer lastId = kwEntitySequenceList.get(0).getSeqId();
+    Optional<KwEntitySequence> kwEntitySequence =
```

Comment:
Looking at this and the previous implementation I feel like we should add an ordering to this query by sequenceId to make sure we are always getting the latest seqId?
I'm mostly thinking this is the kind of query that would trip us up across different DB implementations?
Thoughts?

Reply:
In fact there could be 2 ways: 
1. instead of ordering get just max id from db, however I would consider this as a short term solution which continues being buggy. The problem is that it could happen that between retrieving the last id from DB and inserting a new one there could be another klaw client inserting the next id which will be larger than the max retrieved.
2. To overcome the situation described in 1. there could be either CAS or sequences being used.
2.1 CAS (compare and swap). First there should be a constraint on db level which could guarantee that these ids are always unique. In that case any attempt to insert something with existing id will fail. So on java level there should be kind of a loop trying to perform the same operation until success.
2.2 Another approach: use DB sequence. in that case there is no need for that logic on java side. Sequence will be increased automatically on db level since it is designed for that.
https://www.postgresql.org/docs/current/sql-createsequence.html
https://dev.mysql.com/doc/refman/8.0/en/example-auto-increment.html

Reply:
yeah I think ideally we should update the tables using the kwsequence to auto increment, something like the below. We might need to do that in a seperate PR though, as we basically take away the adding of the primary key from Klaw and transfer it to the DB (YAY!!!)
![image](https://github.com/Aiven-Open/klaw/assets/121855584/a7883136-28a5-4e87-97e7-1cb17efeb585)

Reply:
the tricky part with sequence is that there should be a special migration step/script during Klaw upgrade.
Since it has to set the right initial sequence value for current setup

Reply:
Well we have a migration Utility that can do that, however, the sequences are used by Clusters, Environments and Teams, would we be able to just alter those tables to auto increment their own PKs ? or would that require a drop table and recreate?

Reply:
in theory no need for dropping tables
the only thing to do is calculation of correct initial value which could be different for different Klaw setup

Reply:
cool lets chat about this offline and see if we can come up with the best way to do this!